unikraft/target/target.go

```diff
@@ -74,8 +74,14 @@ type TargetConfig struct {
 	command []string
 }
 
+var _ fmt.Stringer = (*TargetConfig)(nil)
+
 type Targets []TargetConfig
 
+func (tc TargetConfig) String() string {
+	return fmt.Sprintf("%s (%s)", tc.name, tc.ArchPlatString())
+}
+
```

Comment:
In this PR, the returned value from `String()` is actually the "human-readable" version of the target name, and not the "canonical name". The canonical name is actually `<name>_<plat>-<arch>`, as this is the binary name built by Unikraft, and this should be the returned value since it's appropriate across more of the system. Either the human-readable version is kept locally in the `cmd/kraft/run` namespace or the canonical name is the one that's used by implementing this `String` return.

To approach this, I'd also split it into its own commit. I'm wondering also whether it makes sense to propagate the `String`/`fmt.Stringer` interface to `unikraft.Nameable` ? Since then we could then provide `String` representations for all components. They would be implementation-specific of course, still, like this for targets. 

This would then alleviate the use of the wrapper method [`unikraft.TypeNameVersion`](https://github.com/unikraft/kraftkit/blob/staging/unikraft/type.go#L111), which would return an incorrect value if wrapping a `target.Target` input which is `unikraft.Nameable`.

Reply:
I'll keep the UI logic in `cmd` then, to avoid letting the implementation details leak into the UI logic.

Reply:
Ok, I'll add an issue which focuses on providing `Stirng` as an additional mandatory interface in `unikraft.Nameable` and deprecating `unikraft.TypeNameVersion`.

Reply:
See [c9f2aca1..d7ed9336](https://github.com/unikraft/kraftkit/compare/c9f2aca1bf1aabd685e18e65051ae6016a0631d7..d7ed9336b8c74d70e1f3d2a2f9155efd86dc9872)

Reply:
Issue rendered in #268